### ISSUE TITLE
 `COLLATE` can only be directly followed the column type when executing `create table` 

### ISSUE THREAD

**Issue by meixiaofei** [pingcap/tidb#9468]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
Throw exception while excute the following sql, but it works well under mysql
```sql
CREATE TABLE `test`.`Untitled`  (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `name` varchar(50) NOT NULL DEFAULT '' COLLATE utf8_general_ci,
  PRIMARY KEY (`id`)
);
```
And if change the statement order, TiDB can excute correctly
```sql
CREATE TABLE `test`.`Untitled`  (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `name` varchar(50) COLLATE utf8_general_ci NOT NULL DEFAULT '' ,
  PRIMARY KEY (`id`)
);
```

2. What did you expect to see?

The first type of statement can be also excuted correctly

3. What did you see instead?
![image](https://user-images.githubusercontent.com/20218180/53401772-6a001980-39eb-11e9-869b-1c616d529b70.png)

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
5.7.10-TiDB-v3.0.0-beta-17-g35fce93

**Comment by zimulala** [comment]
@meixiaofei 
Thanks for your report!
Now the `COLLATE` is placed after the type in TiDB. We may not deal with this issue in the short term.

**Comment by zimulala** [comment]
Related issues #5857.

**Comment by zimulala** [comment]
It fixed by #9947.
